File: packages/sdk/output/python/src/testsdk/services/connections.py
from urllib.parse import quote
from ..net import query_serializer
from .base import BaseService


class Connections(BaseService):
    def connections_controller_handle_callback(
        self, location: str, code: str, state: str
    ):
        url_endpoint = "/connections/oauth/callback"
        headers = {}
        query_params = []
        self._add_required_headers(headers)
        if not state:
            raise ValueError("Parameter state is required, cannot be empty or blank.")
        query_params.append(
            query_serializer.serialize_query("form", False, "state", state)
        )
        if not code:
            raise ValueError("Parameter code is required, cannot be empty or blank.")
        query_params.append(
            query_serializer.serialize_query("form", False, "code", code)
        )
        if not location:
            raise ValueError(
                "Parameter location is required, cannot be empty or blank."
            )
        query_params.append(
            query_serializer.serialize_query("form", False, "location", location)
        )
        final_url = self._url_prefix + url_endpoint + "?" + "&".join(query_params)
        res = self._http.get(final_url, headers, True)
        return res

    def connections_controller_get_connections(self):
        url_endpoint = "/connections"
        headers = {}
        self._add_required_headers(headers)

        final_url = self._url_prefix + url_endpoint
        res = self._http.get(final_url, headers, True)
        return res

File: packages/sdk/output/typescript/src/services/connections/Connections.ts
import BaseService from '../../BaseService';

import { serializeQuery } from '../../http/QuerySerializer';

export class ConnectionsService extends BaseService {
  async connectionsControllerHandleCallback(
    state: string,
    code: string,
    location: string,
  ): Promise<any> {
    if (state === undefined || code === undefined || location === undefined) {
      throw new Error(
        'The following are required parameters: state,code,location, cannot be empty or blank',
      );
    }
    const queryParams: string[] = [];
    if (state) {
      queryParams.push(serializeQuery('form', true, 'state', state));
    }
    if (code) {
      queryParams.push(serializeQuery('form', true, 'code', code));
    }
    if (location) {
      queryParams.push(serializeQuery('form', true, 'location', location));
    }
    const urlEndpoint = '/connections/oauth/callback';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}?${queryParams.join('&')}`);
    const response: any = await this.httpClient.get(
      finalUrl,
      {},
      {
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }

  async connectionsControllerGetConnections(): Promise<any> {
    const urlEndpoint = '/connections';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}`);
    const response: any = await this.httpClient.get(
      finalUrl,
      {},
      {
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }
}
